Tests/Unit/MSFT_xWebApplication.Tests.ps1

```diff
@@ -0,0 +1,293 @@
+﻿$DSCModuleName = 'xWebAdministration'
+$DSCResourceName = 'MSFT_xWebApplication'
+
+$Splat = @{
+    Path = $PSScriptRoot
+    ChildPath = "..\..\DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    Resolve = $true
+    ErrorAction = 'Stop'
+}
+$DSCResourceModuleFile = Get-Item -Path (Join-Path @Splat)
+
+# should check for the server OS
+if ($env:APPVEYOR_BUILD_VERSION)
+{
+    Add-WindowsFeature -Name Web-Server -Verbose
+}
+
+if (Get-Module -Name $DSCResourceName)
+{
+    Remove-Module -Name $DSCResourceName
+}
+
+Import-Module -Name $DSCResourceModuleFile.FullName -Force
+
+$ModuleRoot = "${env:ProgramFiles}\WindowsPowerShell\Modules\$DSCModuleName"
+
+if (-not (Test-Path -Path $ModuleRoot -PathType Container))
+{
+    New-Item -Path $ModuleRoot -ItemType Directory | Out-Null
+}
+
+Copy-Item -Path "$PSScriptRoot\..\..\*" -Destination $ModuleRoot -Recurse -Force -Exclude '.git'
+
+
+InModuleScope -ModuleName  $DSCResourceName -ScriptBlock {
+
+    $MockParameters = @{
+        Website       = 'MockSite'
+        Name          = 'MockApp'
+        WebAppPool    = 'MockPool'
+        PhysicalPath  = 'C:\MockSite\MockApp'
+    }
+
+    Describe 'CheckDependencies' {
+
+        Context 'WebAdminstration module is not installed' {
+
+            Mock -CommandName Get-Module -MockWith {
+                return $null
+            }
+
+            It 'should throw an error' {
+                {
+                    CheckDependencies
+                } | Should Throw 'Please ensure that WebAdministration module is installed.'
+            }
+
+        }
+
+    }
+
+
+    Describe 'Get-TargetResource' {
+
+        Context 'Absent should return correctly' {
+
+            Mock -CommandName Get-WebApplication -MockWith {
+                return $null
+            }
+
+            It 'should return Absent' {
+                $Result = Get-TargetResource @MockParameters
+                $Result.Ensure | Should Be 'Absent'
+            }
+
+        }
+
+        Context 'Present should return correctly' {
+
+            Mock -CommandName Get-WebApplication -MockWith {
+                return @{
+                    ApplicationPool = $MockParameters.WebAppPool
+                    PhysicalPath = $MockParameters.PhysicalPath
+                    Count = 1
+                }
+            }
+
+            It 'should return Present' {
+                $Result = Get-TargetResource @MockParameters
+                $Result.Ensure | Should Be 'Present'
+            }
+
+        }
+
+    }
+
+
+    Describe "how Test-TargetResource responds to Ensure = 'Absent'" {
```

Comment:
Could we change this to:
````powershell
Describe "how MSFT_xWebApplication\Test-TargetResource responds to Ensure = 'Absent'"
````
So we know which funciton we're calling where in our ever growing pile of tests?